cachi2/core/package_managers/pip.py

```diff
@@ -273,74 +273,61 @@ def _generate_purl_dependency(package: dict[str, Any]) -> str:
-    :return: Tuple of strings (name, version)
-    """
-    name = None
-    version = None
 
+def _extract_metadata_from_config_files(
+    package_dir: RootedPath,
+) -> tuple[Optional[str], Optional[str]]:
     pyproject_toml = PyProjectTOML(package_dir)
     setup_py = SetupPY(package_dir)
     setup_cfg = SetupCFG(package_dir)
 
+    # pyproject.toml
     if pyproject_toml.exists():
-        log.info("Extracting metadata from pyproject.toml")
+        log.debug("Checking pyproject.toml for metadata")
         if pyproject_toml.check_dynamic_version():
-            log.warning("Parsing dynamic metadata from pyproject.toml is not supported")
+            log.warning("Dynamic version parsing from pyproject.toml is not supported")
 
-        name = pyproject_toml.get_name()
-        version = pyproject_toml.get_version()
+        name, version = pyproject_toml.get_name(), pyproject_toml.get_version()
+        if name and version:
+            return name, version
 
-    if None in (name, version) and setup_py.exists():
-        log.info("Filling in missing metadata from setup.py")
-        name = name or setup_py.get_name()
-        version = version or setup_py.get_version()
+    # setup.py
+    if setup_py.exists():
+        log.debug("Checking setup.py for metadata")
+        name, version = setup_py.get_name(), setup_py.get_version()
+        if name and version:
+            return name, version
 
-    if None in (name, version) and setup_cfg.exists():
-        log.info("Filling in missing metadata from setup.cfg")
-        name = name or setup_cfg.get_name()
-        version = version or setup_cfg.get_version()
+    # setup.cfg
+    if setup_cfg.exists():
+        log.debug("Checking setup.cfg for metadata")
+        return setup_cfg.get_name(), setup_cfg.get_version()
 
-    if not name:
-        log.info("Processing metadata from git repository")
-        try:
-            repo_path = get_repo_id(package_dir.root).parsed_origin_url.path.removesuffix(".git")
-            repo_name = Path(repo_path).name
-            package_subpath = package_dir.subpath_from_root
-
-            resolved_path = Path(repo_name).joinpath(package_subpath)
-            normalized_path = canonicalize_name(str(resolved_path).replace("/", "-"))
-            name = normalized_path.strip("-.")
-        except UnsupportedFeature:
-            raise PackageRejected(
-                reason="Could not take name from the repository origin url",
-                solution=(
-                    "Please specify package metadata in a way that Cachi2 understands"
-                    " (see the docs)\n"
-                    "or make sure that the directory Cachi2 is processing is a git"
-                    " repository with\n"
-                    "an 'origin' remote in which case Cachi2 will infer the package name from"
-                    " the remote url."
-                ),
-                docs=PIP_METADATA_DOC,
-            )
+    return None, None
```

Comment:
This refactor isn't a 1:1 replacement - previously if a name was found it was carried over to other source checks. Here, if you extract name but can't find version your code will resort to returning `(None,None)` which means we'll infer the name of the project from the URL instead, that doesn't sound right.